Sources/Yams/YamlError.swift

```diff
@@ -23,11 +23,11 @@ public enum YamlError: Swift.Error {
     // Used in `yaml_parser_t`
     /// `YAML_READER_ERROR`. Cannot read or decode the input stream.
     ///
-    /// - parameter problem:    Error description.
-    /// - parameter byteOffset: The byte about which the problem occured.
-    /// - parameter value:      The problematic value (-1 is none).
-    /// - parameter yaml:       YAML String which the problem occured while reading.
-    case reader(problem: String, byteOffset: Int, value: Int32, yaml: String)
+    /// - parameter problem: Error description.
+    /// - parameter offset:  The offset about which the problem occured.
```

Comment:
I think it's important to document the unit of the offset here. From my understanding, it's this, but let me know if I'm wrong or if this description can be improved:

> The offset at which the problem occurred, in number of code units in the encoding used by the parser.

Reply:
> in number of code units in the encoding used by the parser.

It is `String.IndexDistance?` since converting byte offset to `String.Index` may fail.
So, it may be not in number of code units in the encoding used by the parser.
🤔 

Currently, `Parser` receives `String` so it should not fail here. But in the future `Parser` may receive `Data`, in which case it may actually fail.

Reply:
How do you think about following?
```diff
-    /// - parameter offset:  The offset about which the problem occured.
+    /// - parameter offset:  The offset from `yaml.startIndex` about which the problem occured.
     /// - parameter value:   The problematic value (-1 is none).
     /// - parameter yaml:    YAML String which the problem occured while reading.
-    case reader(problem: String, offset: Int?, value: Int32, yaml: String)
+    case reader(problem: String, offset: String.IndexDistance?, value: Int32, yaml: String)
```

Reply:
Great! Other than changing "about which" to "at which".

Reply:
I forgot that  'IndexDistance' is deprecated.